![](http://www.kiplinger.com/kipimages/pages/14024.jpg)

# An Extensive Medicare Data Exploration - Healthcare Analysis

## About Medicare 

In the United States, Medicare is a single-payer, national social insurance program administered by the U.S. federal government since 1966. United States Medicare is funded by a payroll tax, premiums and surtaxes from beneficiaries, and general revenue. It provides health insurance for Americans aged 65 and older who have worked and paid into the system through the payroll tax. It also provides health insurance to younger people with some disability status as determined by the Social Security Administration, as well as people with end stage renal disease and amyotrophic lateral sclerosis.

In 2015, Medicare provided health insurance for over **55 million** (46 million people age 65 and older and 9 million younger people). On average, Medicare covers about half of the health care charges for those enrolled. The enrollees must then cover their remaining costs either with supplemental insurance, separate insurance, or out-of-pocket. Out-of-pocket costs can vary depending on the amount of health care a Medicare enrollee needs. 

## About CMS

The Centers for Medicare & Medicaid Services (CMS), previously known as the Health Care Financing Administration (HCFA), is a federal agency within the United States Department of Health and Human Services (HHS) that administers the Medicare program and works in partnership with state governments to administer Medicaid, the Children's Health Insurance Program (CHIP), and health insurance portability standards. In addition to these programs, CMS has other responsibilities, including the administrative simplification standards from the Health Insurance Portability and Accountability Act of 1996 (HIPAA), quality standards in long-term care facilities (more commonly referred to as nursing homes) through its survey and certification process, clinical laboratory quality standards under the Clinical Laboratory Improvement Amendments, and oversight of HealthCare.gov.

## About the Data-Set

This [public dataset](https://www.kaggle.com/cms/cms-medicare) was created by the Centers for Medicare & Medicaid Services. The data summarizes the utilization and payments for procedures, services, and prescription drugs provided to Medicare beneficiaries by specific inpatient and outpatient hospitals, physicians, and other suppliers. The dataset includes the following data: Common inpatient and outpatient services, All physician and other supplier procedures and services, All Part D prescriptions. Providers determine what they will charge for items, services, and procedures provided to patients and these charges are the amount that providers bill for an item, service, or procedure.



## Contents

1. Multiple Regression

&nbsp;&nbsp;&nbsp;&nbsp;  1.1 Model fitting

&nbsp;&nbsp;&nbsp;&nbsp;  1.2 Model interpretation
  





In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt

import warnings
warnings.filterwarnings("ignore")

import pandas as pd
from plotly.offline import init_notebook_mode, iplot
import plotly.express as px

import plotly.graph_objects as go

import statsmodels.formula.api as smf

In [ ]:
nursingFacilities=pd.read_csv('https://raw.githubusercontent.com/thousandoaks/AnalyticsUtilizationManagement/main/data/nursing_facilities_2014.csv')

In [ ]:
nursingFacilities.head()

Unnamed: 0  provider_id                                facility_name  \
0           0       235618          HEARTLAND HEALTH CARE CENTER-CANTON   
1           1       495217          MANORCARE HEALTH SERVICES-FAIR OAKS   
2           2       495121  CHERRYDALE HEALTH AND REHABILITATION CENTER   
3           3       225554                               COOLIDGE HOUSE   
4           4       235663              NOTTING HILL OF WEST BLOOMFIELD   

                       street_address          city state  zip_code  \
0                    7025 LILLEY ROAD        CANTON    MI     48187   
1  12475 LEE JACKSON MEMORIAL HIGHWAY       FAIRFAX    VA     22033   
2                    3710 LEE HIGHWAY     ARLINGTON    VA     22207   
3                   30 WEBSTER STREET     BROOKLINE    MA      2446   
4                     6535 DRAKE ROAD  W BLOOMFIELD    MI     48322   

   total_stays  distinct_beneficiaries_per_provider  \
0         1104                                  829   
1          436                                  350   
2          353                                  289   
3          368                                  297   
4          871                                  697   

   average_length_of_stays_days  ...  percent_of_beneficiaries_with_copd  \
0                          25.1  ...                                40.0   
1                          26.3  ...                                29.0   
2                          23.1  ...                                34.0   
3                          19.5  ...                                16.0   
4                          27.2  ...                                29.0   

   percent_of_beneficiaries_with_depression  \
0                                      51.0   
1                                      43.0   
2                                      49.0   
3                                      55.0   
4                                      50.0   

   percent_of_beneficiaries_with_diabetes  \
0                                    47.0   
1                                    46.0   
2                                    49.0   
3                                    46.0   
4                                    49.0   

   percent_of_beneficiaries_with_hyperlipidemia  \
0                                          70.0   
1                                          72.0   
2                                          67.0   
3                                          68.0   
4                                          74.0   

   percent_of_beneficiaries_with_hypertension  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   

   percent_of_beneficiaries_with_ihd  \
0                               68.0   
1                               54.0   
2                               50.0   
3                               45.0   
4                               67.0   

   percent_of_beneficiaries_with_osteoporosis  \
0                                        20.0   
1                                        17.0   
2                                        14.0   
3                                        25.0   
4                                        23.0   

   percent_of_beneficiaries_with_ra_oa  \
0                                 69.0   
1                                 53.0   
2                                 47.0   
3                                 65.0   
4                                 70.0   

   percent_of_beneficiaries_with_schizophrenia  \
0                                         10.0   
1                                         11.0   
2                                         24.0   
3                                         15.0   
4                                         10.0   

   percent_of_beneficiaries_with_stroke  
0                                    17  
1      

# 1. Multiple Regression

In [ ]:
reg = smf.ols(formula='total_snf_medicare_payment_amount ~ total_stays+average_hcc_score+average_length_of_stays_days', data=nursingFacilities)

In [ ]:
results=reg.fit()

In [ ]:
# Inspect the results
print(results.summary())

                                    OLS Regression Results                                   
Dep. Variable:     total_snf_medicare_payment_amount   R-squared:                       0.894
Model:                                           OLS   Adj. R-squared:                  0.894
Method:                                Least Squares   F-statistic:                 4.236e+04
Date:                               Mon, 20 Mar 2023   Prob (F-statistic):               0.00
Time:                                       16:55:28   Log-Likelihood:            -2.2096e+05
No. Observations:                              15026   AIC:                         4.419e+05
Df Residuals:                                  15022   BIC:                         4.419e+05
Df Model:                                          3                                         
Covariance Type:                           nonrobust                                         
                                   coef    std err          

## Based on the previous we have fitted the following model:

 $total\_snf\_medicare\_payment\_amount=-1.682*10^6+1.149*10^4*total\_stays+1.231*10^5*average\_hcc\_score+4.58*10^4*average\_length\_of\_stays\_days+u $



## How do we interpret the equation?



#### If the total stays increases by one unit, total medicare payment is predicted to change by about 11490 dollars. This means that, on average, each inpatient costs 11490 dollars (controlling for other factors).

#### If the average_hcc_score increases by one unit, total medicare payment is predicted to change by about 123100 dollars. 

#### If the average lenght of stays increases by one unit, total medicare payment is predicted to change by about 45800 dollars. This means that, on average, each day of hospitalization costs 45800 dollars (controlling for other factors).


In [ ]:
results.rsquared

0.8942916992646643

### This is not a bad model, it is able to  explain 89% of the total variation in medicare payments
### Conversely, the current model leaves 11%% of the variation unexplained.
